sonar-db/src/main/java/org/sonar/db/user/RoleDao.java

```diff
@@ -21,17 +21,21 @@
 
 import java.util.List;
 import javax.annotation.Nullable;
-import org.apache.ibatis.session.SqlSession;
 import org.sonar.api.security.DefaultGroups;
 import org.sonar.db.Dao;
 import org.sonar.db.DbSession;
 
 public class RoleDao implements Dao {
 
+  /**
+   * @deprecated replaced by {@link org.sonar.db.permission.UserPermissionDao#selectUserPermissions(DbSession, String, String)}
+   */
+  @Deprecated
   public List<String> selectUserPermissions(DbSession session, String userLogin, @Nullable Long resourceId) {
     return mapper(session).selectUserPermissions(userLogin, resourceId);
   }
 
+  // TODO to be moved to PermissionVerifierDao
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)